pkg/common-controller/snapshot_controller.go

```diff
@@ -457,6 +456,9 @@ func (ctrl *csiSnapshotCommonController) syncUnreadySnapshot(snapshot *crdv1.Vol
 					ctrl.updateSnapshotErrorStatusWithEvent(snapshot, v1.EventTypeWarning, "SnapshotStatusUpdateFailed", fmt.Sprintf("Snapshot status update failed, %v", err))
 					return err
 				}
+
+				msg := fmt.Sprintf("waiting for a snapshot %s to be created by the CSI driver.", snapshot.Name)
+				ctrl.eventRecorder.Event(snapshot, v1.EventTypeNormal, "CreatingSnapshot", msg)
```

Comment:
Can you move this event to inside the createSnapshotContent() right after the content is created successfully?
https://github.com/kubernetes-csi/external-snapshotter/blob/master/pkg/common-controller/snapshot_controller.go#L573

Reply:
https://github.com/kubernetes-csi/external-snapshotter/blob/d11ac31736c46b82af0390d2dfc3d5143c7b0a88/pkg/common-controller/snapshot_controller.go#L447
if "updateSnapshotStatus" exec failed,  it will not update event to "EventTypeWarning"， I think add it in there is suitable.

Reply:
createSnapshotContent() happens on line 439.
On line 447, there is a call to updateSnapshotStatus() which has retries.

We should send event after content is created successfully here: https://github.com/kubernetes-csi/external-snapshotter/blob/master/pkg/common-controller/snapshot_controller.go#L573

We should not wait for updateSnapshotStatus.

Reply:
If we add the event after https://github.com/kubernetes-csi/external-snapshotter/blob/master/pkg/common-controller/snapshot_controller.go#L573 

After "updateSnapshotStatus()"   executed failed,  the "updateSnapshotErrorStatusWithEvent()" can not update the event type.

Reply:
Why is that a problem?  Creating a VolumeSnapshotContent and updating VolumeSnapshot status are two different types of updates.  Once VolumeSnapshotContent is created successfully, the sidecar will start provision the snapshot thru CSI driver.  That's why we need to send an event right after a VolumeSnapshotContent is created.  VolumeSnapshot status update failure will be retried later.

Reply:
![image](https://user-images.githubusercontent.com/13997135/79529349-07f6ed80-809f-11ea-890c-47cfa8a80168.png)
 Like this

Reply:
Can you modify the unit test then?

Reply:
Sure

Reply:
Only modify the expected event to "Normal CreatingSnapshot waiting ....."?

Reply:
That's fine.